src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -171,58 +171,71 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
                 }
 
                 HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-                Conversion conversion =
-                    operand != null
-                    ? this.Conversions.ClassifyConversionFromExpression(operand, patternType, ref useSiteDiagnostics, forCast: true)
-                    : this.Conversions.ClassifyConversionFromType(operandType, patternType, ref useSiteDiagnostics, forCast: true);
+                var matchPossible = ExpressionOfTypeMatchesPatternType(Conversions, operandType, patternType, ref useSiteDiagnostics, out var conversion, operand.ConstantValue, operandCouldBeNull: true);
```

Comment:
I think Aleksey's feedback still applies in latest iteration. Please use bool? instead of var here.

---
In reply to: [116913667](https://github.com/dotnet/roslyn/pull/19413#discussion_r116913667) [](ancestors = 116913667)